src/test/SimpleTestReporter.h

```diff
@@ -36,9 +36,21 @@ struct SimpleTestReporter : public ConsoleReporter
     }
 
     void
-    assertionStarting(AssertionInfo const& ai) override
+    sectionStarting(SectionInfo const& _sectionInfo) override
     {
         printDot();
+        ConsoleReporter::sectionStarting(_sectionInfo);
+    }
+
+    void
+    sectionEnded(SectionStats const& _sectionStats) override
+    {
+        ConsoleReporter::sectionEnded(_sectionStats);
+    }
+
+    void
+    assertionStarting(AssertionInfo const& ai) override
```

Comment:
Perhaps I'm missing something, but this function doesn't seem to do anything anymore. `assertionStarting` does nothing but store the argument in the private variable `mLastAssertInfo`. But the only place `mLastAssertInfo` is used is as the argument to `assertionStarting`. If this is expected, can you explain what is happening here?

Reply:
Why do we no longer want to print a dot on `assertionStarting`?

Reply:
`mLastAssertInfo` is used in `assertionEnded`. The parameter to `Ended` doesn't have the right information (like the line number). In essence, what this reporter does is filter the events that are passed down to the console reporter, which is why we simulate the right sequence of start/end for the console reporter so that it has the right data to do its reporting.

Reply:
dot on `assertionStarting` turned out to be completely useless (almost never triggers), now it's doing it with each section which is much better